Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs

```diff
@@ -61,46 +67,102 @@ private IEnumerable<UpdatedExternalReferenceProperties> MapReferences(IEnumerabl
-                InsecureCountriesSubjectToDataTransferUuids = dto.InsecureCountriesSubjectToDataTransferUuids.FromNullable().AsChangedValue(),
-                DataProcessorUuids = dto.DataProcessorUuids.FromNullable().AsChangedValue(),
-                HasSubDataProcessors = (dto.HasSubDataProcessors?.ToYesNoUndecidedOption()).AsChangedValue(),
-                SubDataProcessorUuids = dto.SubDataProcessorUuids.FromNullable().AsChangedValue()
+                DataResponsibleUuid = rule.MustUpdate(x => x.General.DataResponsibleUuid)
+                    ? dto.DataResponsibleUuid.AsChangedValue()
+                    : OptionalValueChange<Guid?>.None,
+
+                DataResponsibleRemark = rule.MustUpdate(x => x.General.DataResponsibleRemark)
+                    ? dto.DataResponsibleRemark.AsChangedValue()
+                    : OptionalValueChange<string>.None,
+
+                IsAgreementConcluded = rule.MustUpdate(x => x.General.IsAgreementConcluded)
+                    ? (dto.IsAgreementConcluded?.ToYesNoIrrelevantOption()).AsChangedValue()
+                    : OptionalValueChange<YesNoIrrelevantOption?>.None,
+
+                IsAgreementConcludedRemark = rule.MustUpdate(x => x.General.IsAgreementConcludedRemark)
+                    ? dto.IsAgreementConcludedRemark.AsChangedValue()
+                    : OptionalValueChange<string>.None,
+
+                AgreementConcludedAt = rule.MustUpdate(x => x.General.AgreementConcludedAt)
+                    ? dto.AgreementConcludedAt.AsChangedValue()
+                    : OptionalValueChange<DateTime?>.None,
+
+                BasisForTransferUuid = rule.MustUpdate(x => x.General.BasisForTransferUuid)
+                    ? dto.BasisForTransferUuid.AsChangedValue()
+                    : OptionalValueChange<Guid?>.None,
+
+                TransferToInsecureThirdCountries = rule.MustUpdate(x => x.General.TransferToInsecureThirdCountries)
+                    ? (dto.TransferToInsecureThirdCountries?.ToYesNoUndecidedOption()).AsChangedValue()
+                    : OptionalValueChange<YesNoUndecidedOption?>.None,
+
+                InsecureCountriesSubjectToDataTransferUuids =
+                    rule.MustUpdate(x => x.General.InsecureCountriesSubjectToDataTransferUuids)
+                        ? dto.InsecureCountriesSubjectToDataTransferUuids.FromNullable().AsChangedValue()
+                        : OptionalValueChange<Maybe<IEnumerable<Guid>>>.None,
+
+                DataProcessorUuids = rule.MustUpdate(x => x.General.DataProcessorUuids)
+                    ? dto.DataProcessorUuids.FromNullable().AsChangedValue()
+                    : OptionalValueChange<Maybe<IEnumerable<Guid>>>.None,
+
+                HasSubDataProcessors = rule.MustUpdate(x => x.General.HasSubDataProcessors)
+                    ? (dto.HasSubDataProcessors?.ToYesNoUndecidedOption()).AsChangedValue()
```

Comment:
Bliver denne nulstillet hvis man sender "Null" som value her?

Reply:
Ved mig skal jeg lave ?? operator for at null bliver set som en ændring

Reply:
Virker fordi det er nullable value type som target